core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -208,14 +263,30 @@ public HttpResponse doCompleteInstall() throws IOException, ServletException {
         return HttpResponses.okJSON();
     }
     
-    static void completeSetup(Jenkins jenkins) {
+    static void completeSetup(Jenkins jenkins) throws IOException {
         jenkins.setInstallState(InstallState.INITIAL_SETUP_COMPLETED);
         InstallUtil.saveLastExecVersion();
+        UpgradeWizard.completeUpgrade(jenkins);
         // Also, clean up the setup wizard if it's completed
         jenkins.setSetupWizard(null);
     }
     
     /**
+     * Handle security configuration
+     */
+    public HttpResponse doSaveSecurityConfig(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, FormException {
+        jenkins.checkPermission(Jenkins.ADMINISTER); // this is redundant, maybe
+        GlobalSecurityConfiguration securityConfig = (GlobalSecurityConfiguration)jenkins.getDynamic("configureSecurity");
+        boolean configureSuccess = securityConfig.configure(req, req.getSubmittedForm());
+        if(configureSuccess) {
+            jenkins.save();
+            jenkins.setInstallState(InstallState.CONFIGURE_SECURITY.getNextState());
+            return HttpResponses.okJSON();
+        }
+        return HttpResponses.errorJSON("Unable to configure security");
```

Comment:
@svanoort not really sure there's anything we can do in this case. this code wasn't changed, however just indented